rust/otap-dataflow/crates/otap/src/encoder/error.rs

```diff
@@ -0,0 +1,15 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! this module contains definitions for the various errors that can happen encoding OTAP data
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("An arrow error occurred encoding error record batch: {0}")]
+    ArrowError(#[from] arrow::error::ArrowError),
+
+    #[error("An error occured serializing value as CBOR: {0}")]
+    CborError(#[from] ciborium::ser::Error<std::io::Error>),
```

Comment:
By integrating errors from our dependencies into our public errors, we leak implementation details. For example, if we change the crate for CBOR serialization/deserialization in the future, we would have to change the signature of the `CborError` variant.
In general, I try to avoid this by stringifying errors from our dependencies and adding context when it makes sense.

Reply:
good call, made this change